numpy/core/src/multiarray/multiarraymodule.c

```diff
@@ -3199,7 +3170,13 @@ array_arange(PyObject *NPY_UNUSED(ignored),
 NPY_NO_EXPORT unsigned int
 PyArray_GetNDArrayCVersion(void)
 {
-    return (unsigned int)NPY_ABI_VERSION;
+    // return (unsigned int)NPY_ABI_VERSION;
+    /*
+     * TODO: Preliminary returning the 1.x API version, that is a lie but
+     *       allows (for the moment) downstream modules to mix and match
+     *       and us to import old matplotlib versions in our doc builds...
+     */
+    return (unsigned int)0x01000009;
```

Comment:
Since I put it at the wrong place first:

This is clearly an evil hack.  I think I am leaning towards doing it though (with an issue opened of course!).  It cannot be in a final release of course, and we could escalate to requiring an env variable soon.

Now, of course there is a good chance this will break down earlier anyway, because we will have enough reason to create that `numpy2_compat` package (meaning the de-facto ABI compatibility breaks down).

@mattip and @rgommers what do you think?  Or is this too piece-meal for you?

Reply:
I am perfectly fine with such piecemeal tweaks to get to the desired end state. I think we can expect the C ABI to break multiple times during the 2.0 journey, and it's okay to lie a bit or do whatever is pragmatic to keep our CI or nightlies in a functional state during this period.